plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/IcebergQueryRunner.java

```diff
@@ -46,64 +51,70 @@ public static DistributedQueryRunner createIcebergQueryRunner()
         return createIcebergQueryRunner(
                 Map.of(),
                 Map.of(),
-                List.of());
+                Optional.empty());
     }
 
     public static DistributedQueryRunner createIcebergQueryRunner(
             Map<String, String> extraProperties,
             Map<String, String> connectorProperties,
-            Iterable<TpchTable<?>> tables)
+            Optional<File> metastoreDirectory)
             throws Exception
     {
         return createIcebergQueryRunner(
                 extraProperties,
                 connectorProperties,
-                tables,
-                Optional.empty());
+                metastoreDirectory,
+                IcebergQueryRunner::initializeSchema);
     }
 
     public static DistributedQueryRunner createIcebergQueryRunner(
             Map<String, String> extraProperties,
             Map<String, String> connectorProperties,
-            Iterable<TpchTable<?>> tables,
-            Optional<File> metastoreDirectory)
+            Optional<File> metastoreDirectory,
+            Consumer<QueryRunner> schemaInitializer)
             throws Exception
     {
         Session session = testSessionBuilder()
                 .setCatalog(ICEBERG_CATALOG)
-                .setSchema("tpch")
+                .setSchema(SCHEMA)
                 .build();
 
         DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(session)
                 .setExtraProperties(extraProperties)
                 .build();
 
         queryRunner.installPlugin(new TpchPlugin());
-        queryRunner.createCatalog("tpch", "tpch");
-
-        Path dataDir = metastoreDirectory.map(File::toPath).orElseGet(() -> queryRunner.getCoordinator().getBaseDataDir().resolve("iceberg_data"));
+        queryRunner.createCatalog("tpch", SCHEMA);
 
         queryRunner.installPlugin(new IcebergPlugin());
         connectorProperties = new HashMap<>(ImmutableMap.copyOf(connectorProperties));
-        connectorProperties.putIfAbsent("iceberg.catalog.type", "TESTING_FILE_METASTORE");
-        connectorProperties.putIfAbsent("hive.metastore.catalog.dir", dataDir.toString());
+        if (!connectorProperties.containsKey("iceberg.catalog.type")) {
```

Comment:
Should this belong to "Allow passing schema initializer to Iceberg query runner" commit?

Reply:
Possible completely separate commit. But it's a cosmetic change so i thought as method signature was changed in whole util, this fix may be applied as well.
Will make a separate commit.